src/gnu/trove/array/AbstractOffheapArray.java

```diff
@@ -53,11 +60,41 @@ public AbstractOffheapArray(long capacity) {
         }
         address = UNSAFE.allocateMemory(capacity);
         boxedAddress = new AtomicLong(address);
-        cleaner = Cleaner.create(this, new Deallocator(boxedAddress));
+        cleaner = createCleaner(this, new Deallocator(boxedAddress));
         UNSAFE.setMemory(address, capacity, (byte) 0);
         this.capacity = capacity;
     }
 
+    private static Object createCleaner(Object obj, Runnable cleanUpRunnable) {
+        try {
+            Class<?> cleaner = Class.forName("sun.misc.Cleaner");
+            Method create = cleaner.getMethod("create", Object.class, Runnable.class);
```

Comment:
Can we memoize these and avoid reflective lookups when each object is constructed?

Reply:
Yeah, I guess it isn't that bad to just keep them around, so I'll go ahead and do that